### ISSUE TITLE
Pairwise part-to-part predictions

### ISSUE THREAD

**Issue by lauritk** [DeepLabCut/DeepLabCut#219]
Hi,
noticed that pairwise part-to-part prediction is not incorporated in the DeepLabCut, like it was in the original DeeperCut paper. Is there a particular reason for not to use pairwise probabilities to improve predictions?

**Comment by AlexEMG** [comment]
Yes, we took it out for simplicity, _speed_ and as the accuracy for the examples in our [NN paper](https://www.nature.com/articles/s41593-018-0209-y) already reaches human level accuracy. There are certainly some cases, where it is useful and I am actually planning to add it back in (as a fully independent model).

**Comment by lauritk** [comment]
Thank you for prompt reply. Generally the accuracy and overall performance are really good. I experienced few corner cases while I'm training model to track rat/rodent in open field test for my CS master's thesis. We have lots of old and heterogeneous video material from open field type settings and noticed that with some of the videos the selection of canditates could have been more restricted. Some body parts were erratically jumping all over the place or switching back and forth with other detected parts. This happened especially with soft, low resolution and bad quality records with Sprague Dawley on white background. Not an easy task for AI, but it worked surprisingly well with more clearer cases.

If I understood the DeeperCut paper correctly, pairwise probabilities could restrict plausible poses of the body?

**Comment by AlexEMG** [comment]
Pairwise could help for jumping, however, in my experience jumping of body parts is due to some errors in the labeling data (flipped label, etc.). Even very low contrast cases (e.g. check out right figure in case study 5: http://www.mousemotorlab.org/deeplabcutdetails) the feature detectors are enough (you might just need e.g. more training data). Pairwise for such a situation seems to be overkill.

**Comment by lauritk** [comment]
Thank you for the tips. Will double check the training data for errors. The training set is already quite big, but maybe I have to add more frames from low contrast videos.

**Comment by AlexEMG** [comment]
_but maybe I have to add more frames from low contrast videos_ >> I think so.
